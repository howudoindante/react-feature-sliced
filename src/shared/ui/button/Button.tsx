import { type ButtonHTMLAttributes, type FC } from 'react';
import s from './Button.module.scss';
import classClue from '../../lib/classclue';
interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    className?: string;
}
export const Button: FC<ButtonProps> = (props) => {
    const { className, children, ...otherProps } = props;
    return (
        <button {...otherProps} className={classClue([s.button, className])}>
            {children}
        </button>
    );
};
