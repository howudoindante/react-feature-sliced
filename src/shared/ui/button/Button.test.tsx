import renderer from 'react-test-renderer';
import { Button } from './Button';
it('Classname appends right', async () => {
    const component = renderer.create(<Button type="button">Test</Button>);
    let tree = component.toJSON() as renderer.ReactTestRendererJSON;

    expect(tree).toMatchSnapshot();

    await renderer.act(() => {
        component.update(
            <Button type="button" className="test">
                Test
            </Button>
        );
    });
    tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

    await renderer.act(() => {
        component.update(<Button type="button">Test</Button>);
    });
    // re-rendering
    tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
});
