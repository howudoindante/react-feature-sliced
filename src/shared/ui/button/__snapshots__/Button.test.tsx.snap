// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classname appends right 1`] = `
<button
  className="button"
  type="button"
>
  Test
</button>
`;

exports[`Classname appends right 2`] = `
<button
  className="button test"
  type="button"
>
  Test
</button>
`;

exports[`Classname appends right 3`] = `
<button
  className="button"
  type="button"
>
  Test
</button>
`;
